src/encoding/hybrid_rle/mod.rs

```diff
@@ -54,7 +54,7 @@ fn read_next<'a, 'b>(decoder: &'b mut Decoder<'a>, remaining: usize) -> Result<S
             });
             let value = u32::from_le_bytes(bytes);
             if additional == 1 {
-                State::Single(value)
+                State::Single(Some(value))
```

Comment:
Because we have this small number optimization, the stop iteration call indicated by returning `None` was never triggered.

By keeping a `Option<u32>` state we keep track of this and return `None` once a value is already taken.